changelog.d/13614.feature

```diff
@@ -0,0 +1 @@
+Support setting the registration shared secret in a file, via a new `registration_shared_secret_path` configuration option.
```

Comment:
This feels very similar to me to simply putting the `registration_shared_secret` option in a separate configuration file -- can you describe a bit more why this is useful?

Reply:
Well, it's particularly useful in combination with #13615. With both PRs in place, I can have a k8s deployment which creates a homeserver.yaml with `registration_shared_secret_path` pointing to a non-existent file. Then Synapse will magically generate me a shared secret on first run (which `register_new_matrix_user` can slurp).

Reply:
Will that cause the shared secret to be regenerated each time the container is rebuilt? I'm struggling to see why you would want to generate these on first run.

Reply:
> Will that cause the shared secret to be regenerated each time the container is rebuilt? 

That depends if you use a persistent volume to keep `registration_shared_secret_path` on. If you do, you'll keep the same secret between rebuilds. But it doesn't matter anyway: there is little benefit to using the same shared secret on each run, provided `register_new_matrix_user` can find the current secret. Indeed, maybe it's better to rotate it frequently.

>  I'm struggling to see why you would want to generate these on first run.

... well, because if you don't have synapse generate them, you have to generate them yourself with some annoying scripting.